hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/BlockCache.java

```diff
@@ -279,4 +280,14 @@ default boolean isCacheEnabled() {
   default boolean waitForCacheInitialization(long timeout) {
     return true;
   }
+
+  /**
+   * Allows for BlockCache implementations to provide a mean to refresh their configurations. Since
+   * HBASE-28517, CacheConfig implements ConfigurationObserver and registers itself for
+   * notifications of dynamic configuration changes. The default is a noop.
+   * @param config the new configuration to be updated.
+   */
+  default void refreshConfiguration(Configuration config) {
```

Comment:
Why not just implement ConfigurationObserver?

Reply:
I just thought to reuse CacheConfig as the observer for all cache related configuration changes notifications. Sounds less convoluted to have just one observer, but no hard feelings about it. I can change if you prefer to have BlockCache implementing ConfigurationObserver.

Reply:
We can make CacheConfig implement PropagatingConfigurationObserver, so it can register children ConfigurationObservers to ConfigurationManager.

Reply:
Ok, have pushed a new commit with these suggestions.